### PR TITLE
Adding KMS key attribute to SNS topic resource

### PR DESCRIPTION
**Pull request by Rohit1509**
### Description

Adding KMS key attribute to SNS topic resource

### Issues Resolved

Adding new attribute so no issues resolved

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: test/unit/resources/mock/iam/aws_sns_topic_mock.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def stub_data
+    stub_data = []
+
+    topic = {:client => Aws::SNS::Client,
+             :method => :get_topic_attributes,
+             :data => @topic}
+
+    stub_data += [topic]
```
Comment (sathish-progress) [inline test/unit/resources/mock/iam/aws_sns_topic_mock.rb:20]: why not simply?

```suggestion
    topic = {
      client: Aws::SNS::Client,
      method: :get_topic_attributes,
      data: @topic
    }
    [topic]
```

Comment (Rohit1509) [reply test/unit/resources/mock/iam/aws_sns_topic_mock.rb:20]: Since for rest of the resources, this method was used. I followed similar procedure.

Comment (sathish-progress) [reply test/unit/resources/mock/iam/aws_sns_topic_mock.rb:20]: Okay, If this makes better sense, could you update it..?  Old resources may need some refactoring in the future..

Comment (Rohit1509) [reply test/unit/resources/mock/iam/aws_sns_topic_mock.rb:20]: Definitely, I will make that commit in a minute.

Comment (sathish-progress) [reply test/unit/resources/mock/iam/aws_sns_topic_mock.rb:20]: you could commit the suggestion directly if you would like to ..

Comment (Rohit1509) [reply test/unit/resources/mock/iam/aws_sns_topic_mock.rb]: committed necessary changes

**Review by sathish-progress — APPROVED** [review]
LGTM

**Pull request merged**